pt_miniscreen/core/components/list.py

```diff
@@ -85,49 +114,51 @@ def _scroll_transition(self):
                 self.state.update({"transition_progress": progress + progress_step})
 
         active_transition = self.state["active_transition"]
-        row_to_remove = self.rows[0 if active_transition == "DOWN" else -1]
-        self.rows.remove(row_to_remove)
-        self.remove_child(row_to_remove)
+        if active_transition == "DOWN":
+            rows_to_remove = self.rows[:distance]
+        else:
+            rows_to_remove = self.rows[-distance:]
+
+        for row in rows_to_remove:
+            self.rows.remove(row)
+            self.remove_child(row)
         self._rows_snapshot = None
         self.state.update({"active_transition": None, "transition_progress": 0})
 
-    def scroll_up(self):
+    def scroll_to(self, direction, distance=1):
         if self.state["active_transition"] is not None:
-            logger.debug(f"{self} currently scrolling, ignoring scroll up")
-            return
-
-        next_top_row_index = self.state["top_row_index"] - 1
-        if next_top_row_index < 0:
-            logger.debug(f"{self} has no more rows, ignoring scroll up")
+            logger.info(f"{self} currently scrolling, ignoring scroll")
             return
 
-        Row = self.state["Rows"][next_top_row_index]
-        self.rows.insert(0, self.create_child(Row))
+        if direction == "UP":
+            if not self.can_scroll_up:
```

Comment:
When scrolling with a distance greater than one it's possible to get an out of range exception since this assumes the scroll distance will be one. If it took the distance as an argument this would work well :)

```
        if direction == "UP":
            if not self.can_scroll_up(distance):
```